### PR TITLE
[MEN-5342] Integrate the lifecycle workflow with the new API

### PR DESCRIPTION
**Pull request by alfrunes**

### CODE REVIEW

File: client/iothub/model.go [hunk 1]
```diff
@@ -88,6 +88,15 @@ const (
 	StatusDisabled Status = "disabled"
 )
 
+func NewStatusFromMenderStatus(status model.Status) Status {
```
Comment (MaciejTe) [inline client/iothub/model.go:91]: Nitpick: please consider differentiating Mender device status from Azure device status

File: app/app.go [hunk 2]
```diff
@@ -120,11 +112,23 @@ func (a *app) GetDeviceIntegrations(
...
+		integrations, err := a.store.GetIntegrations(ctx,
+			model.IntegrationFilter{IDs: device.IntegrationIDs},
+		)
+		return integrations, errors.Wrap(err, "app: failed to get device integrations")
+	}
 	return []model.Integration{}, nil
 }
 
-func (a *app) SetDeviceStatus(ctx context.Context, deviceID string, status Status) error {
+func (a *app) SetDeviceStatus(ctx context.Context, deviceID string, status model.Status) error {
```
Comment (MaciejTe) [inline app/app.go:131]: Nitpick: please differentiate Mender device status from Azure device status: 
```
SetDeviceStatus(ctx context.Context, deviceID string, menderStatus model.Status)
```

**Review by MaciejTe — APPROVED** [review]
Looks good!

**Review by tranchitella — APPROVED** [review]

File: client/iothub/model.go (hunk 1 above)
Comment (tranchitella) [reply client/iothub/model.go:91]: @MaciejTe feel free to address this in your PR

File: app/app.go (hunk 2 above)
Comment (tranchitella) [reply app/app.go:131]: @MaciejTe feel free to address this in your PR

**Pull request merged**